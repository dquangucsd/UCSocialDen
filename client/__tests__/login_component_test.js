import 'react-native';
import React from 'react';
import LoginScreen from '../app/login';
import { render, waitFor, fireEvent } from '@testing-library/react-native';
import { useRouter } from "expo-router";
import AsyncStorage from "@react-native-async-storage/async-storage";

jest.mock("@react-native-async-storage/async-storage", () => ({
  getItem: jest.fn(),
}));

describe('LoginScreen', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<LoginScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  beforeEach(() => {
      jest.clearAllMocks();
  });

  test("shows login button when no token is found", async () => {
      AsyncStorage.getItem.mockResolvedValue(null);

      const { getByText } = render(<LoginScreen />);

      await waitFor(() => {
          expect(getByText("Login with UCSD Email")).toBeTruthy();
      });
  });
});