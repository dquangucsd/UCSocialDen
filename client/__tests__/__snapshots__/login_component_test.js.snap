// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../assets/images/login.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    pointerEvents="none"
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0.3)",
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "position": "relative",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#556ebe",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 16,
          "zIndex": 2,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Zain",
            "fontSize": 34,
            "fontWeight": "bold",
            "zIndex": 2,
          }
        }
      >
        UC Social Den
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Zain",
            "fontSize": 50,
            "fontWeight": "bold",
            "marginBottom": 30,
            "marginTop": 200,
            "zIndex": 2,
          }
        }
      >
        Welcome to UC Social Den
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#4285F4",
            "borderRadius": 10,
            "opacity": 1,
            "paddingHorizontal": 20,
            "paddingVertical": 20,
            "zIndex": 2,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "zIndex": 2,
            }
          }
        >
          Login with UCSD Email
        </Text>
      </View>
    </View>
  </View>
</View>
`;
